### ISSUE TITLE
Git Sync configuration to automatically stage and commit before sync

### ISSUE THREAD

**Issue by cybarspace** [microsoft/vscode#97230]
Is there any way to configure the "Synchronization" icon on the statusbar to stage and commit (with a commit message template) before it syncs ?
.
I don't want to have to write my own extension for this, but it'd be nice if I could just edit the settings (GUI or JSON) to achieve this effect.
.
![image](https://user-images.githubusercontent.com/50134239/81388624-5b0e0e80-913a-11ea-8695-4426a982506f.png)

**Comment by cybarspace** [comment]
Ok, found a "hacky" fix.

1. Install the [multi-command](https://marketplace.visualstudio.com/items?itemName=ryuta46.multi-command) extension.

2. Install the [Shortcuts](https://marketplace.visualstudio.com/items?itemName=gizak.shortcuts) extension.

3. Edit settings.json to add the following lines :
```
"multiCommand.commands": [
    {
        "command": "multiCommand.gitSync",
        "sequence": [
            "workbench.action.files.saveAll",
            "git.stageAll",
            "git.commitStaged",
            "git.sync"
        ]
    }
],
"shortcuts.buttons": [
    "triangle-up , multiCommand.gitSync , Git Sync"
]
```
4. Restart VS Code.

This adds a triangle-up icon to the status bar which does stage, commit, pull, push in a single click. It does however ask for a commit message, which I'd be better off without (using a commit template). But it works quite well otherwise.

I'll look into how I can use a commit message template. I'll close this issue after I solve this completely, unless admins see it fit to close it now.

**Comment by joaomoreno** [comment]
I'm glad you found a workaround for this! Let me know how the commit template goes.

**Comment by cybarspace** [comment]
@joaomoreno More workarounds :
I made a `commitmsg.txt` file in the root of the local repo I'm working on and edited it to contain the commit template. Then executed the following command in Terminal (with repository root as working directory) : 
`git config commit.template commitmsg.txt`
And it seems to work fine now. Doesn't ask for commit message anymore.
.
Weird thing though. I actually did this _before_ using the multi-command and Shortcut extension. But maybe something with the Shortcut extension wasn't acting right. Now I use [Activitus Bar](https://marketplace.visualstudio.com/items?itemName=Gruntfuggly.activitusbar) extension. Had to add the following to `settings.json`, overwriting the config I had to do for Shortcut extension : 
```
"activitusbar.views": [
    {
        "name": "command.multiCommand.gitSync",
        "codicon": "arrow-up",
        "tooltip": "Git Sync"
    }
]
```
Kept the multi-command config same though. Now it works with the commit template.

I wonder why Shortcuts extension couldn't handle it automatically but Activitus Bar could.
But I'm glad I can work much smoother now. I'll close the issue now. Thanks. :smiley:
